libcontainer/configs/validate/validator.go

```diff
@@ -70,6 +71,45 @@ func rootfs(config *configs.Config) error {
 	return nil
 }
 
+// https://elixir.bootlin.com/linux/v6.12/source/net/core/dev.c#L1066
+func devValidName(name string) bool {
+	if len(name) == 0 || len(name) > unix.IFNAMSIZ {
+		return false
+	}
+	if (name == ".") || (name == "..") {
+		return false
+	}
+	if strings.ContainsAny(name, "/: ") {
+		return false
+	}
+	return true
+}
+
+func netdevices(config *configs.Config) error {
+	if len(config.NetDevices) == 0 {
+		return nil
+	}
+	if !config.Namespaces.Contains(configs.NEWNET) {
+		return errors.New("unable to move network devices without a NET namespace")
+	}
+	if config.Namespaces.IsPrivate(configs.NEWNET) {
+		return errors.New("unable to move network devices without a NET namespace")
+	}
```

Comment:
This seems to be wrong. This checks if it has a `Path` set. But this should work for any network namespace.

When you remove this, the rest of the code might fail if you need the netns fd. In that case, I think you should get the ns fs from `/proc/<pid>/ns/net`, or even better if you can use the pidfd as that doesn't have races and those pains.

IMHO, we should implement both: use pidfd whenever possible and fallback to `/proc/<pid>/ns/net` when that is not available, and handle the races properly.